tests/test_dmm.py

```diff
@@ -0,0 +1,167 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import cast
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+
+from pulser.channels.dmm import DMM
+from pulser.register.base_register import BaseRegister
+from pulser.register.mappable_reg import MappableRegister
+from pulser.register.register_layout import RegisterLayout
+from pulser.register.weight_maps import DetuningMap
+
+
+@pytest.fixture
+def trap_coordinates():
+    return [[0, 0], [1, 0], [0, 1], [1, 1]]
+
+
+@pytest.fixture
+def layout(trap_coordinates) -> RegisterLayout:
+    return RegisterLayout(trap_coordinates)
+
+
+@pytest.fixture
+def register(layout: RegisterLayout) -> BaseRegister:
+    return layout.define_register(0, 1, 2, 3, qubit_ids=(0, 1, 2, 3))
+
+
+@pytest.fixture
+def map_reg(layout: RegisterLayout) -> MappableRegister:
+    return layout.make_mappable_register(4)
+
+
+def test_init(
+    layout: RegisterLayout, register: BaseRegister, map_reg: MappableRegister
+):
+    str_key = {"1": 1.0}
+    wrong_key = {4: 1.0}
+    bad_weights = {0: -1.0, 1: 1.0, 2: 1.0}
+    bad_sum = {0: 0.1, 2: 0.9, 3: 0.1}
+    det_map = {0: 0.7, 1: 0.3, 2: 0}
+    slm_map = {0: 1 / 3, 1: 1 / 3, 2: 1 / 3}
+    for bad_key in (str_key, wrong_key):
+        for reg in (layout, map_reg):
+            with pytest.raises(
+                ValueError,
+                match=(
+                    "The trap ids of detuning weights have to be integers"
+                    " between 0 and 4"
+                ),
+            ):
+                reg.define_detuning_map(bad_key)  # type: ignore
+        with pytest.raises(
+            ValueError,
+            match=(
+                "The qubit ids of detuning weights have to be defined in the"
+                " register."
+            ),
+        ):
+            register.define_detuning_map(bad_key)  # type: ignore
+    with pytest.raises(
+        ValueError, match="Number of traps and weights don't match."
+    ):
+        DetuningMap([(0, 0), (1, 0)], [0])
+    for reg in (layout, map_reg, register):
+        with pytest.raises(
+            ValueError, match="All weights must be non-negative."
+        ):
+            reg.define_detuning_map(bad_weights)  # type: ignore
+        with pytest.raises(
+            ValueError, match="The sum of the weights should be 1."
+        ):
+            reg.define_detuning_map(bad_sum)  # type: ignore
+        for map in (det_map, slm_map):
+            detuning_map = cast(
+                DetuningMap, reg.define_detuning_map(map)  # type: ignore
+            )
+            assert np.all(
+                [map[i] == detuning_map.weights[i] for i in range(len(map))]
+            )
+            assert np.all(
+                [
+                    layout.coords[i]
+                    == np.array(detuning_map.trap_coordinates)[i]
+                    for i in range(len(map))
+                ]
+            )
```

Comment:
Since you're defining fixtures for `det_map` and `slm_map` anyway, why not put this in a separate test function and use them?
Or even use them here, if you'd prefer that